pod/annotations.go

```diff
@@ -626,3 +626,54 @@ func SidecarAnnotations(sidecarName, suffix string) []string {
 		LegacySidecarAnnotation(sidecarName, suffix),
 	}
 }
+
+type PlatformSidecar struct {
+	Name     string
+	Channel  string
+	ArgsJSON []byte
+}
+
+// PlatformSidecars parses sidecar-related annotations and returns a structured
+// slice of platform sidecars.
+func PlatformSidecars(annotations map[string]string) []PlatformSidecar {
+	// Look for legacy include annotation
+	var sidecarNames []string
+	if sidecarList := annotations[AnnotationKeySidecarsIncludeLegacy]; sidecarList != "" {
+		sidecarNames = strings.Split(sidecarList, " ")
+	}
+
+	// Handle the current include annotation format
+	for a, val := range annotations {
+		if strings.HasSuffix(a, "."+AnnotationKeySuffixSidecars) {
+			if boolVal, _ := strconv.ParseBool(val); boolVal {
+				sidecarNames = append(sidecarNames, strings.TrimSuffix(a, "."+AnnotationKeySuffixSidecars))
+			}
+		}
+	}
+
+	var sidecars []PlatformSidecar
+	for _, sidecarName := range sidecarNames {
+		sidecar := PlatformSidecar{
+			Name:    sidecarName,
+			Channel: "default",
+		}
+		for _, key := range SidecarAnnotations(sidecarName, "channel") {
```

Comment:
Nit: probably not necessary to have a separate method if this is the only place to use it. Though I'm okay with leaving it as is too, since it's about to be removed once we deprecate the legacy format anyway.

Reply:
Agreed, I've removed the function and inlined here.